src/util/address.rs

```diff
@@ -215,6 +215,11 @@ impl Address {
         Address(self.0 - size)
     }
 
+    // Perform a saturating subtract on the Address
+    pub const fn saturating_sub(self, size: usize) -> Address {
```

Comment:
This function seems unsafe to me because it creates an address from an integer, same as `Address::from_usize`.

Reply:
Hm... Isn't that true for `Address::sub` just above it as well?

Reply:
Oh, sorry I misunderstood what it does.
Still, I'm not sure about the safety of `Address::sub`, but it's fine for now